### PR TITLE
Correct non-existent exception in ec2_asg

### LINKED ISSUE #2526: Nonexistent exception in ec2_asg.py
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

ec2_elb
##### ANSIBLE VERSION

N/A
##### SUMMARY

The EC2 autoscaling module tries to catch a nonexist exception `boto.exception.InvalidInstance` at https://github.com/ansible/ansible-modules-core/commit/b2df875e78d6be12409bf7a3926c005b23993170#diff-689b2e5ce09076c21c9915e50f8a9140R292.  This should be handled like https://github.com/ansible/ansible-modules-core/blob/devel/cloud/amazon/ec2_elb.py#L243 instead.

### PR DESCRIPTION
**Pull request by shawnsi**
Fixes #2526

### CODE REVIEW

**Comment by robynbergeron** [comment]
Hi @shawnsi -- thanks for submitting this PR.

@garethr -- Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.
If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

**Comment by gregdek** [comment]
Thanks @shawnsi for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by shawnsi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @shawnsi. @garethr please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Thanks @shawnsi. @garethr please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
@garethr This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by john-dodson** [comment]
@robynbergeron @garethr can someone please merge this? Functionality is completely broken for me and I'd rather not run production deployments off someone's forked branch.

**Comment by gregdek** [comment]
@garethr still waiting on your review.  Please comment with text 'shipit' or 'needs_revision' as appropriate. If we don't hear from you within 14 days, we will start to look for additional maintainers for this module.

**Comment by gregdek** [comment]
Moving into review by the @ansible/core team. @shawnsi apologies for the delay.

**Comment by shawnsi** [comment]
I've updated the exception handling per this [comment](https://github.com/ansible/ansible-modules-core/issues/2526#issuecomment-181404347) in the issue.

**Comment by helsing** [comment]
@gregdek Is there any progress with this pull request?

**Comment by gregdek** [comment]
@shawnsi @helsing this PR has been moved into core_review because there are currently no community maintainers. Would one/both of you be willing to become maintainers for this module?

**Comment by shawnsi** [comment]
@gregdek Are you still looking to identify a maintainer?  This change is pretty simple and would be very helpful if merged.

**Comment by gregdek** [comment]
I'm gonna put this one into shipit and see what happens. :)

File: cloud/amazon/ec2_asg.py [hunk 1]
```diff
@@ -333,8 +333,12 @@ def elb_healthy(asg_connection, elb_connection, module, group_name):
         # but has not yet show up in the ELB
         try:
             lb_instances = elb_connection.describe_instance_health(lb, instances=instances)
-        except boto.exception.InvalidInstance:
-            pass
+        except boto.exception.BotoServerError, e:
```
Comment (nitzmahone) [inline cloud/amazon/ec2_asg.py]: Just to keep things happy on the python3 front, this should be `except boto.exception.BotoServerError as e:`

**Comment by nitzmahone** [comment]
@shawnsi One minor nit on this one- ping me when it's fixed and I'll merge. Thanks!

**Comment by shawnsi** [comment]
@nitzmahone Its fixed.  Thanks for the prompt review.

**Comment by nitzmahone** [comment]
Noice- thanks!

**Pull request merged**